src/main/java/com/weather/service/WeatherService.java

```diff
@@ -19,19 +21,29 @@
 
     private final RestTemplate restTemplate;
 
-    public WeatherService(RestTemplate restTemplate) {
+    private final WeatherRepository weatherRepository;
+
+    private final WeatherForecastRepository weatherForecastRepository;
+
+    public WeatherService(RestTemplate restTemplate, WeatherRepository weatherRepository, WeatherForecastRepository weatherForecastRepository) {
         this.restTemplate = restTemplate;
+        this.weatherRepository = weatherRepository;
+        this.weatherForecastRepository = weatherForecastRepository;
     }
 
     public Weather getWeather(String city,String country)
     {
         URI url=new UriTemplate(ApplicationConstants.WEATHER_URL).expand(city,country,ApplicationConstants.API_KEY);
+        Weather weatherObj=invoke(url,Weather.class);
+        weatherRepository.save(weatherObj);
         return invoke(url,Weather.class);
```

Comment:
Can you justify, why two times of calling of invoke() method?

Reply:
First time the invoke() method is returning the object to be saved in repository.
UPDATION: I will the weatherObj instead of again calling the invoke() method